synapse/lib/autodoc.py

```diff
@@ -46,14 +47,26 @@
                            'documentation for.',
             'additionalProperties': False,
         },
-
+        'deprecatedItem': {
+            'type': 'object',
+            'properties': {
+                'eolvers': {'type': ['string', 'null'], 'default': 'v3.0.0',
+                            'description': "The version which will not longer support the item."},
+                'eoldate': {'type': ['string', 'null'], 'default': None,
```

Comment:
Maybe `eolvers` and `eoldate` should be a `oneOf` so you can specify one of them but not both? Does it make sense to have both a version and a date?

Reply:
Doesn't make sense to allow both values.